data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/BulkRetryStrategy.java

```diff
@@ -152,16 +160,46 @@ public static boolean canRetry(final Exception e) {
                         !NON_RETRY_STATUS.contains(((OpenSearchException) e).status().getStatus())));
     }
 
+    private void handleRetriesAndFailures(final AccumulatingBulkRequest bulkRequestForRetry,
+                                          final int retryCount,
+                                          final BackOffUtils backOffUtils,
+                                          final BulkResponse bulkResponse,
+                                          Exception e) throws InterruptedException {
+        boolean doRetry = (Objects.isNull(e)) ? canRetry(bulkResponse) : canRetry(e);
+        if (!Objects.isNull(bulkResponse) && doRetry && retryCount == 1) { // first attempt
```

Comment:
Why does `doRetry` need to be true here? If it is `false` and some documents were successfully sent on the first attempt we should increment the counter for those documents.

Reply:
@cmanning09 I guess you are right. I just refactored the existing code. It looks like even existing code is incrementing stat only when `canRetry` is true. Modified the code to increment the stat even when the doRetry is not true.